packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -242,11 +265,8 @@ class ConnectionStateHandler implements IConnectionStateHandler {
             },
         );
 
-        // Based on recent data, it looks like majority of cases where we get stuck are due to really slow or
-        // timing out ops fetches. So attempt recovery infrequently. Also fetch uses 30 second timeout, so
-        // if retrying fixes the problem, we should not see these events.
         this.joinOpTimer = new Timer(
-            JoinOpTimeoutMs,
+            JoinOpTimeoutMs, // default value is not used - startJoinOpTimer() explicitly provides timeout
```

Comment:
FYI on the existence of TimerWithNoDefaultTimeout.  Currently lives in garbageCollection.ts but now that there's two spots that could use it, maybe it can move somewhere.  Or you may opt to just user Timer directly without enforcement that start is passed a timeout. In my case I wanted to make sure all callers specified the timeout.

Reply:
Also if you leave it as-is I'd pass 0, it'll make the bug clearer if someone doesn't pass a value to start.